File: src/extras-kservice/libkservicegroup.cpp
#include <KServiceGroup>
#include <KSycocaEntry>
#include <QList>
#include <QSharedData>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kservicegroup.h>
#include "libkservicegroup.h"
#include "libkservicegroup.hxx"

KServiceGroup* KServiceGroup_new(const libqt_string name) {
    QString name_QString = QString::fromUtf8(name.data, name.len);
    return new KServiceGroup(name_QString);
}

KServiceGroup* KServiceGroup_new2(const libqt_string _fullpath, const libqt_string _relpath) {
    QString _fullpath_QString = QString::fromUtf8(_fullpath.data, _fullpath.len);
    QString _relpath_QString = QString::fromUtf8(_relpath.data, _relpath.len);
    return new KServiceGroup(_fullpath_QString, _relpath_QString);
}

libqt_string KServiceGroup_RelPath(const KServiceGroup* self) {
    QString _ret = self->relPath();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KServiceGroup_Caption(const KServiceGroup* self) {
    QString _ret = self->caption();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KServiceGroup_Icon(const KServiceGroup* self) {
    QString _ret = self->icon();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KServiceGroup_Comment(const KServiceGroup* self) {
    QString _ret = self->comment();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

int KServiceGroup_ChildCount(const KServiceGroup* self) {
    return self->childCount();
}

bool KServiceGroup_NoDisplay(const KServiceGroup* self) {
    return self->noDisplay();
}

bool KServiceGroup_ShowEmptyMenu(const KServiceGroup* self) {
    return self->showEmptyMenu();
}

void KServiceGroup_SetShowEmptyMenu(KServiceGroup* self, bool b) {
    self->setShowEmptyMenu(b);
}

bool KServiceGroup_ShowInlineHeader(const KServiceGroup* self) {
    return self->showInlineHeader();
}

void KServiceGroup_SetShowInlineHeader(KServiceGroup* self, bool _b) {
    self->setShowInlineHeader(_b);
}

bool KServiceGroup_InlineAlias(const KServiceGroup* self) {
    return self->inlineAlias();
}

void KServiceGroup_SetInlineAlias(KServiceGroup* self, bool _b) {
    self->setInlineAlias(_b);
}

bool KServiceGroup_AllowInline(const KServiceGroup* self) {
    return self->allowInline();
}

void KServiceGroup_SetAllowInline(KServiceGroup* self, bool _b) {
    self->setAllowInline(_b);
}

int KServiceGroup_InlineValue(const KServiceGroup* self) {
    return self->inlineValue();
}

void KServiceGroup_SetInlineValue(KServiceGroup* self, int _val) {
    self->setInlineValue(static_cast<int>(_val));
}

libqt_list /* of libqt_string */ KServiceGroup_SuppressGenericNames(const KServiceGroup* self) {
    QList<QString> _ret = self->suppressGenericNames();
    // Convert QList<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * (_ret.size() + 1)));
    for (qsizetype i = 0; i < _ret.size(); ++i) {
        QString _lv_ret = _ret[i];
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _lv_b = _lv_ret.toUtf8();
        libqt_string _lv_str;
        _lv_str.len = _lv_b.length();
        _lv_str.data = static_cast<const char*>(malloc(_lv_str.len + 1));
        memcpy((void*)_lv_str.data, _lv_b.data(), _lv_str.len);
        ((char*)_lv_str.data)[_lv_str.len] = '\0';
        _arr[i] = _lv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

void KServiceGroup_SetLayoutInfo(KServiceGroup* self, const libqt_list /* of libqt_string */ layout) {
    QList<QString> layout_QList;
    layout_QList.reserve(layout.len);
    libqt_string* layout_arr = static_cast<libqt_string*>(layout.data);
    for (size_t i = 0; i < layout.len; ++i) {
        QString layout_arr_i_QString = QString::fromUtf8(layout_arr[i].data, layout_arr[i].len);
        layout_QList.push_back(layout_arr_i_QString);
    }
    self->setLayoutInfo(layout_QList);
}

libqt_list /* of libqt_string */ KServiceGroup_LayoutInfo(const KServiceGroup* self) {
    QList<QString> _ret = self->layoutInfo();
    // Convert QList<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * (_ret.size() + 1)));
    for (qsizetype i = 0; i < _ret.size(); ++i) {
        QString _lv_ret = _ret[i];
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _lv_b = _lv_ret.toUtf8();
        libqt_string _lv_str;
        _lv_str.len = _lv_b.length();
        _lv_str.data = static_cast<const char*>(malloc(_lv_str.len + 1));
        memcpy((void*)_lv_str.data, _lv_b.data(), _lv_str.len);
        ((char*)_lv_str.data)[_lv_str.len] = '\0';
        _arr[i] = _lv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

libqt_string KServiceGroup_BaseGroupName(const KServiceGroup* self) {
    QString _ret = self->baseGroupName();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KServiceGroup_DirectoryEntryPath(const KServiceGroup* self) {
    QString _ret = self->directoryEntryPath();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void KServiceGroup_Delete(KServiceGroup* self) {
    delete self;
}

File: include/extras-kservice/libkservicegroup.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const ksycocatype_enums = @import("libksycocatype.zig").enums;
const std = @import("std");

/// https://api.kde.org/kservicegroup.html
pub const kservicegroup = struct {
    /// New constructs a new KServiceGroup object.
    ///
    /// ``` name: []const u8 ```
    pub fn New(name: []const u8) QtC.KServiceGroup {
        const name_str = qtc.libqt_string{
            .len = name.len,
            .data = name.ptr,
        };

        return qtc.KServiceGroup_new(name_str);
    }

    /// New2 constructs a new KServiceGroup object.
    ///
    /// ``` _fullpath: []const u8, _relpath: []const u8 ```
    pub fn New2(_fullpath: []const u8, _relpath: []const u8) QtC.KServiceGroup {
        const _fullpath_str = qtc.libqt_string{
            .len = _fullpath.len,
            .data = _fullpath.ptr,
        };
        const _relpath_str = qtc.libqt_string{
            .len = _relpath.len,
            .data = _relpath.ptr,
        };

        return qtc.KServiceGroup_new2(_fullpath_str, _relpath_str);
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#relPath)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn RelPath(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KServiceGroup_RelPath(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.RelPath: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#caption)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn Caption(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KServiceGroup_Caption(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.Caption: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#icon)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn Icon(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KServiceGroup_Icon(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.Icon: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#comment)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn Comment(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KServiceGroup_Comment(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.Comment: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#childCount)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn ChildCount(self: ?*anyopaque) i32 {
        return qtc.KServiceGroup_ChildCount(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#noDisplay)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn NoDisplay(self: ?*anyopaque) bool {
        return qtc.KServiceGroup_NoDisplay(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#showEmptyMenu)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn ShowEmptyMenu(self: ?*anyopaque) bool {
        return qtc.KServiceGroup_ShowEmptyMenu(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#setShowEmptyMenu)
    ///
    /// ``` self: QtC.KServiceGroup, b: bool ```
    pub fn SetShowEmptyMenu(self: ?*anyopaque, b: bool) void {
        qtc.KServiceGroup_SetShowEmptyMenu(@ptrCast(self), b);
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#showInlineHeader)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn ShowInlineHeader(self: ?*anyopaque) bool {
        return qtc.KServiceGroup_ShowInlineHeader(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#setShowInlineHeader)
    ///
    /// ``` self: QtC.KServiceGroup, _b: bool ```
    pub fn SetShowInlineHeader(self: ?*anyopaque, _b: bool) void {
        qtc.KServiceGroup_SetShowInlineHeader(@ptrCast(self), _b);
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#inlineAlias)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn InlineAlias(self: ?*anyopaque) bool {
        return qtc.KServiceGroup_InlineAlias(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#setInlineAlias)
    ///
    /// ``` self: QtC.KServiceGroup, _b: bool ```
    pub fn SetInlineAlias(self: ?*anyopaque, _b: bool) void {
        qtc.KServiceGroup_SetInlineAlias(@ptrCast(self), _b);
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#allowInline)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn AllowInline(self: ?*anyopaque) bool {
        return qtc.KServiceGroup_AllowInline(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#setAllowInline)
    ///
    /// ``` self: QtC.KServiceGroup, _b: bool ```
    pub fn SetAllowInline(self: ?*anyopaque, _b: bool) void {
        qtc.KServiceGroup_SetAllowInline(@ptrCast(self), _b);
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#inlineValue)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn InlineValue(self: ?*anyopaque) i32 {
        return qtc.KServiceGroup_InlineValue(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#setInlineValue)
    ///
    /// ``` self: QtC.KServiceGroup, _val: i32 ```
    pub fn SetInlineValue(self: ?*anyopaque, _val: i32) void {
        qtc.KServiceGroup_SetInlineValue(@ptrCast(self), @intCast(_val));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#suppressGenericNames)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn SuppressGenericNames(self: ?*anyopaque, allocator: std.mem.Allocator) [][]const u8 {
        const _arr: qtc.libqt_list = qtc.KServiceGroup_SuppressGenericNames(@ptrCast(self));
        const _str: [*]qtc.libqt_string = @ptrCast(@alignCast(_arr.data));
        defer {
            for (0.._arr.len) |i| {
                qtc.libqt_string_free(@ptrCast(&_str[i]));
            }
            qtc.libqt_free(_arr.data);
        }
        const _ret = allocator.alloc([]const u8, _arr.len) catch @panic("kservicegroup.SuppressGenericNames: Memory allocation failed");
        for (0.._arr.len) |i| {
            const _data = _str[i];
            const _buf = allocator.alloc(u8, _data.len) catch @panic("kservicegroup.SuppressGenericNames: Memory allocation failed");
            @memcpy(_buf, _data.data[0.._data.len]);
            _ret[i] = _buf;
        }
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#setLayoutInfo)
    ///
    /// ``` self: QtC.KServiceGroup, layout: [][]const u8, allocator: std.mem.Allocator ```
    pub fn SetLayoutInfo(self: ?*anyopaque, layout: [][]const u8, allocator: std.mem.Allocator) void {
        var layout_arr = allocator.alloc(qtc.libqt_string, layout.len) catch @panic("kservicegroup.SetLayoutInfo: Memory allocation failed");
        defer allocator.free(layout_arr);
        for (layout, 0..layout.len) |item, i| {
            layout_arr[i] = .{
                .len = item.len,
                .data = item.ptr,
            };
        }
        const layout_list = qtc.libqt_list{
            .len = layout.len,
            .data = layout_arr.ptr,
        };
        qtc.KServiceGroup_SetLayoutInfo(@ptrCast(self), layout_list);
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#layoutInfo)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn LayoutInfo(self: ?*anyopaque, allocator: std.mem.Allocator) [][]const u8 {
        const _arr: qtc.libqt_list = qtc.KServiceGroup_LayoutInfo(@ptrCast(self));
        const _str: [*]qtc.libqt_string = @ptrCast(@alignCast(_arr.data));
        defer {
            for (0.._arr.len) |i| {
                qtc.libqt_string_free(@ptrCast(&_str[i]));
            }
            qtc.libqt_free(_arr.data);
        }
        const _ret = allocator.alloc([]const u8, _arr.len) catch @panic("kservicegroup.LayoutInfo: Memory allocation failed");
        for (0.._arr.len) |i| {
            const _data = _str[i];
            const _buf = allocator.alloc(u8, _data.len) catch @panic("kservicegroup.LayoutInfo: Memory allocation failed");
            @memcpy(_buf, _data.data[0.._data.len]);
            _ret[i] = _buf;
        }
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#baseGroupName)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn BaseGroupName(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KServiceGroup_BaseGroupName(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.BaseGroupName: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#directoryEntryPath)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn DirectoryEntryPath(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KServiceGroup_DirectoryEntryPath(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.DirectoryEntryPath: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#isType)
    ///
    /// ``` self: QtC.KServiceGroup, t: ksycocatype_enums.KSycocaType ```
    pub fn IsType(self: ?*anyopaque, t: i32) bool {
        return qtc.KSycocaEntry_IsType(@ptrCast(self), @intCast(t));
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#sycocaType)
    ///
    /// ``` self: QtC.KServiceGroup ```
    ///
    /// Returns: ``` ksycocatype_enums.KSycocaType ```
    pub fn SycocaType(self: ?*anyopaque) i32 {
        return qtc.KSycocaEntry_SycocaType(@ptrCast(self));
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#name)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn Name(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KSycocaEntry_Name(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.Name: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#entryPath)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn EntryPath(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KSycocaEntry_EntryPath(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.EntryPath: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#storageId)
    ///
    /// ``` self: QtC.KServiceGroup, allocator: std.mem.Allocator ```
    pub fn StorageId(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KSycocaEntry_StorageId(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kservicegroup.StorageId: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#isValid)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.KSycocaEntry_IsValid(@ptrCast(self));
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#isDeleted)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn IsDeleted(self: ?*anyopaque) bool {
        return qtc.KSycocaEntry_IsDeleted(@ptrCast(self));
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#setDeleted)
    ///
    /// ``` self: QtC.KServiceGroup, deleted: bool ```
    pub fn SetDeleted(self: ?*anyopaque, deleted: bool) void {
        qtc.KSycocaEntry_SetDeleted(@ptrCast(self), deleted);
    }

    /// Inherited from KSycocaEntry
    ///
    /// [Qt documentation](https://api.kde.org/ksycocaentry.html#isSeparator)
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn IsSeparator(self: ?*anyopaque) bool {
        return qtc.KSycocaEntry_IsSeparator(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kservicegroup.html#dtor.KServiceGroup)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KServiceGroup ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KServiceGroup_Delete(@ptrCast(self));
    }
};

/// https://api.kde.org/kservicegroup.html#types
pub const enums = struct {
    pub const EntriesOption = enum {
        pub const NoOptions: i32 = 0;
        pub const SortEntries: i32 = 1;
        pub const ExcludeNoDisplay: i32 = 2;
        pub const AllowSeparators: i32 = 4;
        pub const SortByGenericName: i32 = 8;
    };
};
